zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/DataStoreManager.scala

```diff
@@ -161,12 +161,29 @@ class DataStoreManager(metaDataset: String,
     val dropTableName = dropTable.dataset
 
     if(metaData.contains(dropTableName)){
+
       metaData.remove(dropTableName)
+      context.child("data-" + dropTableName).foreach( child => child ! PoisonPill)
+      val metaRecordFilter = FilterStatement(DataSetInfo.MetaSchema.fieldMap("name"), None, Relation.matches, Seq(dropTableName))
+      metaActor ! DeleteRecord(metaDataset, Seq(metaRecordFilter))
+
+      metaData.filter{ p =>
```

Comment:
we'd better use `case (name, info) => ` instead of `p` to avoid unamed `_1` and `_2` which has little information.